### ISSUE TITLE
Transforming string for compatibility with vcf format

### ISSUE THREAD

**Issue by henrikstranneheim** [Clinical-Genomics/GeneLists#21]
Problem: VCF format does not allow “ “, in key=value pairs. We internally use “,”, “|” and other characters to split records and some annotated string contain these characters.
Solution: Transforming the string to a list of unicode code point

Method:

1. split unicode object to individual characters and store in list
2. Use python functions ord() and chr() to switch between original string and unicodes code point representation (https://docs.python.org/3/library/functions.html#ord)
3. join code points or characters with “|”  or “” as delimiter for print/storing in database

Here is some pseudo code:
Example:

Gene_description=“ADK:Hello world” from Ensembl query

String -> Unicode code point
1. A, D, K, :, H, e, l, l, o,  , w, o., r, l, d
2. 65, 68, 75, 58, 72, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100
3. 65|68|75|58|72|101|108|108|111|32|119|111|114|108|100 - This is saved in the list(s) for the gene entry

Gene_description=65|68|75|58|72|101|108|108|111|32|119|111|114|108|100
Unicode code point -> String
1. 65, 68, 75, 58, 72, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100
2. A, D, K, :, H, e, l, l, o,  , w, o, r, l, d
3. ADK:Hello world

**Comment by ingkebil** [comment]
This would only be for Gene_description

**Comment by henrikstranneheim** [comment]
Yes, but if we are going through with Måns plan for "get_ids" this will not be necessary.

**Comment by ingkebil** [comment]
True. Well, that's going to be an easy issue to be closed then.
